R/qenv-eval_code.R

```diff
@@ -61,28 +63,30 @@ setMethod("eval_code", signature = c("qenv", "expression"), function(object, cod
         invokeRestart("muffleMessage")
       }
     )
+
     if (!is.null(x)) {
       return(x)
     }
-
-    object@warnings <- c(object@warnings, current_warnings)
-    object@messages <- c(object@messages, current_messages)
   }
+
+
+  object@warnings <- c(object@warnings, current_warnings)
+  object@messages <- c(object@messages, current_messages)
+
   lockEnvironment(object@env, bindings = TRUE)
   object
 })
 
 #' @rdname eval_code
 #' @export
 setMethod("eval_code", signature = c("qenv", "language"), function(object, code) {
-  code_char <- as.expression(code)
-  eval_code(object, code_char)
+  eval_code(object, code = remove_enclosing_curly_braces(as.character(as.expression(code))))
```

Comment:
Theoretically this line could remain unchanged

```suggestion
  eval_code(object, code = as.expression(code))
```

Reply:
clever